src/test/scala/NumberFormatTest.scala

```diff
@@ -0,0 +1,119 @@
+  }
+
+  test("padding using format test") {
+    val numberFormatObj = NumberFormat(Some("$*x#,##0.00"))
+    assert(numberFormatObj.parse("$xx123.00") === 123)
+  }
+
+  test("significant digits test 1") {
+    val numberFormatObj = NumberFormat(Some("@##"))
+    assert(numberFormatObj.format(0.1203) === "0.12")
+  }
+
+  test("should parse numbers that match %000 correctly") {
+    val numberFormatObj = NumberFormat(Some("%000"))
+    assert(numberFormatObj.parse("%001").doubleValue() === 0.01)
+    assert(numberFormatObj.parse("%123").doubleValue() === 1.23)
+    assert(numberFormatObj.parse("%1234").doubleValue() === 12.34)
+  }
+
+  test("should parse numbers that contain ‰ (per-mille) correctly") {
+    // This document says even if the pattern is not provided, implementations should recognise and parse numbers that
+    // consist of ‰ which is not true for NumberFormat class. Check this if some problems occur in future.
+    // https://www.w3.org/TR/2015/REC-tabular-data-model-20151217/#formats-for-numeric-types
+    val numberFormatObj = NumberFormat(Some("‰000"))
+    assert(numberFormatObj.parse("‰1000") === 1)
+  }
+
+  test("should correctly parse numbers in non-english-convention when Group and Decimal chars are configured to that convention") {
+    // English convention: 5,246.30
+    // Non-English convention: 5.246,30
+    // https://en.wikipedia.org/wiki/Decimal_separator#Hindu-Arabic_numerals
+    
+    val numberFormatObjWithConfiguration = NumberFormat(Some("#,##0.00"), Some('.'), Some(','))
+    assert(numberFormatObjWithConfiguration.parse("36.756,32").doubleValue() === 36756.32)
+  }
+
+  test("should not parse numbers in non-english-convention when Group and Decimal chars are standard") {
+    val numberFormatObjWithoutConfiguration = NumberFormat(Some("#,##0.00"), None, None)
+    assert(numberFormatObjWithoutConfiguration.parse("36.756,32").doubleValue() != 36756.32)
+  }
+
+  test("should correctly format numbers in non-english-convention when Group and Decimal chars are configured to that convention") {
+    val numberFormatObjWithConfiguration = NumberFormat(Some("#,##0.00"), Some('.'), Some(','))
+    assert(numberFormatObjWithConfiguration.format(36756.32) === "36.756,32")
+  }
+
+  test("should parse numbers when pattern is not provided") {
+    val numberFormatObj = NumberFormat(None)
+    assert(numberFormatObj.parse("3456.12345").doubleValue() === 3456.12345)
+  }
+
+  test("should format numbers when pattern is not provided") {
+    val numberFormatObj = NumberFormat(None)
+    assert(numberFormatObj.format(3456.12345) === "3,456.123")
+  }
+
+  test("should throw NumberFormatError when the pattern provided is not valid") {
+    val thrown = intercept[NumberFormatError] {
+      NumberFormat(Some("0.#00#"))
+    }
+    assert(thrown.getMessage === "Malformed pattern for ICU DecimalFormat: \"0.#00#\": 0 cannot follow # after decimal point at position 3")
```

Comment:
This test will be a bit brittle. When the third party library changes anything about this exception message, the test will fail. I would say it's enough to assert that there is an exception. Don't assert that it must exactly match that string.

Reply:
Removed checking the exception message, now it just checks if an exception is thrown